components/request_otr/browser/request_otr_tab_storage.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_TAB_STORAGE_H_
+#define BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_TAB_STORAGE_H_
+
+#include "base/memory/ref_counted.h"
+#include "brave/components/brave_shields/browser/blocked_domain_1pes_lifetime.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+using brave_shields::BlockedDomain1PESLifetime;
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace ephemeral_storage {
+class EphemeralStorageService;
+}  // namespace ephemeral_storage
+
+namespace request_otr {
+
+// A short-lived, per tab storage for Request-OTR interstitials, that stores a
+// flag while proceeding so a new interstitial is not shown immediately,
+// and stores the result of the user's request.
+class RequestOTRTabStorage
+    : public content::WebContentsUserData<RequestOTRTabStorage> {
+ public:
+  ~RequestOTRTabStorage() override;
+
+  // Disallow copy and assign.
+  RequestOTRTabStorage(const RequestOTRTabStorage&) = delete;
+  RequestOTRTabStorage& operator=(const RequestOTRTabStorage&) = delete;
+
+  // Returns the RequestOTRTabStorage associated to |web_contents|, or
+  // creates one if there is none.
+  static RequestOTRTabStorage* GetOrCreate(content::WebContents* web_contents);
+
+  void SetIsProceeding(bool is_proceeding) { is_proceeding_ = is_proceeding; }
```

Comment:
what does `IsProceeding` indicate? Please add comments for these. Also the pattern for simple getter/setter is lowercase underscore `set_is_proceeding`/`is_proceeding`

Reply:
I think it's better in general to add comments to the method declarations

Reply:
changing to `set_is_proceeding/is_proceeding` can be a follow-up

Reply:
tracked in https://github.com/brave/brave-browser/issues/30306